src/components/views/settings/TchapJoinRuleSettings.tsx

```diff
@@ -0,0 +1,330 @@
+    }, {
+        value: JoinRule.Public,
+        label: _t("Public"),
+        description: <>
+            { _t("Anyone can find and join.") }
+            { aliasWarning }
+        </>,
+    }];
+
+    if (roomSupportsRestricted || preferredRestrictionVersion || joinRule === JoinRule.Restricted) {
+        let upgradeRequiredPill;
+        if (preferredRestrictionVersion) {
+            upgradeRequiredPill = <span className="mx_JoinRuleSettings_upgradeRequired">
+                { _t("Upgrade required") }
+            </span>;
+        }
+
+        let description;
+        if (joinRule === JoinRule.Restricted && restrictedAllowRoomIds?.length) {
+            // only show the first 4 spaces we know about, so that the UI doesn't grow out of proportion there are lots.
+            const shownSpaces = restrictedAllowRoomIds
+                .map(roomId => cli.getRoom(roomId))
+                .filter(room => room?.isSpaceRoom())
+                .slice(0, 4);
+
+            let moreText;
+            if (shownSpaces.length < restrictedAllowRoomIds.length) {
+                if (shownSpaces.length > 0) {
+                    moreText = _t("& %(count)s more", {
+                        count: restrictedAllowRoomIds.length - shownSpaces.length,
+                    });
+                } else {
+                    moreText = _t("Currently, %(count)s spaces have access", {
+                        count: restrictedAllowRoomIds.length,
+                    });
+                }
+            }
+
+            const onRestrictedRoomIdsChange = (newAllowRoomIds: string[]) => {
+                if (!arrayHasDiff(restrictedAllowRoomIds || [], newAllowRoomIds)) return;
+
+                if (!newAllowRoomIds.length) {
+                    setContent({
+                        join_rule: JoinRule.Invite,
+                    });
+                    return;
+                }
+
+                setContent({
+                    join_rule: JoinRule.Restricted,
+                    allow: newAllowRoomIds.map(roomId => ({
+                        "type": RestrictedAllowType.RoomMembership,
+                        "room_id": roomId,
+                    })),
+                });
+            };
+
+            const onEditRestrictedClick = async () => {
+                const restrictedAllowRoomIds = await editRestrictedRoomIds();
+                if (!Array.isArray(restrictedAllowRoomIds)) return;
+                if (restrictedAllowRoomIds.length > 0) {
+                    onRestrictedRoomIdsChange(restrictedAllowRoomIds);
+                } else {
+                    onChange(JoinRule.Invite);
+                }
+            };
+
+            /**
+             * TCHAP : disable space-related options if create space feature is not enabled
+             */
+            if (ComponentVisibilityCustomisations.shouldShowComponent(UIComponent.CreateSpaces)) {
```

Comment:
I wish we could contribute this to element-web, but it makes sense only for a project without spaces, which element doesn't allow. 
(in the element-web situation, if spaces creation is disabled, the user can still have spaces, and chose the visibility of the room in these spaces.)